src/viking.js

```diff
@@ -1,11 +1,105 @@
-// Soldier
-class Soldier {}
+// Soldier j
+class Soldier {
+	constructor(health,strength){
+	this.health= health;
+	this.strength= strength;
+	}
+	attack(){
+		return this.strength
+	}
+	receiveDamage(damage){
+		this.health-=damage;
+	}
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+	constructor(name,health,strength){
+		super(health,strength);
+		this.name=name;
+	}
+	attack(){
+		return this.strength;
+	}
+
+	receiveDamage(damage)
+	{
+		this.health-=damage;
+	 if(this.health>0)
+	 {
+		return `${this.name} has received ${damage} points of damage`;
+	 } 
+	 else  if(this.health<=0)
+	 {
+		return `${this.name} has died in act of combat`;
+	 }
+	}
+	
+	battleCry()
+	{
+		return "Odin Owns You All!"
+	}
+	
+}
 
 // Saxon
-class Saxon {}
+class Saxon  extends Soldier{
+	receiveDamage(damage)
+	{
+		this.health-=damage;
+	 if(this.health>0)
+	 {
+		return `A Saxon has received ${damage} points of damage`;
+	 } 
+	 else  if(this.health<=0)
+	 {
+		return `A Saxon has died in combat`;
+	 }
+	}
+
+}
 
 // War
-class War {}
+class War {
+	constructor(){
+		this.vikingArmy=[];
+	this.saxonArmy=[]}
+
+	addViking(vikingObject){
+		this.vikingArmy.push(vikingObject);
+	}
+
+	addSaxon(saxonObject){
+		this.saxonArmy.push(saxonObject);
+	}
+
+	vikingAttack(){
+		const randomSaxonDamage = Math.floor((Math.random()*this.saxonArmy.length));
+    const randomVikingDamage = Math.floor((Math.random()*this.vikingArmy.length));
+		const alive=this.saxonArmy[randomSaxonDamage].receiveDamage( this.vikingArmy[randomVikingDamage].strength);
+		if(alive == "A saxon has died in combat"){
+			this.saxonArmy.splice(randomSaxonDamage);
```

Comment:
Aqui funcionaria si comparas health con strength y usar tu metodo splice o pop algo asi:

if(saxon.health <= viking.strength){
            this.saxonArmy.pop(saxon)
        }
        return saxon.receiveDamage(viking.strength)
Aplica lo mismo para sacar del arreglo a vikingos o saxons, si lo intentas me platicas como queda.